### ISSUE TITLE
Bug in Board Hash trait implementation

### ISSUE THREAD

**Issue by barakugav** [jordanbray/chess#75]
The implementation of the Hash trait uses only the field 'hash' which is only affected by the pieces positions. On the other hand, the implementation of PartialEq compare all fields of the struct.
This causes unnecessary collisions in hashmap/hashset.
To solve it, the implementation of the Hash trait could simply use the get_hash() func.

**Comment by jordanbray** [comment]
I do not believe this is true.  The reason the hash trait is implemented the way it is is to incrementally update the hash function.  This is a technique called Zobrist hashing.

If you can find an example of such a collision, I'd love to see it.

**Comment by barakugav** [comment]
The current implementation does not include the castling right for exmaple, or en passant value

**Comment by jordanbray** [comment]
Ahh.  You are correct.  I misread the comment.
